extension.js

```diff
@@ -112,9 +113,49 @@ function description(app) {
     return appName + ' → ' + app.get_title();
 }
 
-function updateHighlight(boxes) {
-  boxes.forEach(box => box.whole.remove_style_class_name('switcher-highlight'));
-  boxes.length > cursor && boxes[cursor].whole.add_style_class_name('switcher-highlight');
+function updateHighlight(boxes, query) {
+  boxes.forEach(box => {
+    // Remove previous selection highlight
+    box.whole.remove_style_class_name('switcher-highlight');
+
+    // Don't highlight description if there's no input
+    if (query == "")
+      return;
+
+    // Identify substring parts to be highlighted
+    let queryExpression = "(";
+    let text = box.label.get_text();
+    let queries = query.split(' ');
+    let queriesLength = queries.length;
+    for (let i = 0; i < queriesLength - 1; i++) {
+      if (queries[i] != "") {
+        queryExpression += queries[i] + "|";
+      }
+    }
+    queryExpression += queries[queriesLength - 1] + ")";
+    let queryRegExp = new RegExp(queryExpression, "i");
+    let tokenRegExp = new RegExp("^" + queryExpression + "$", "i");
+
+    // Build description from highlighted and non-highlighted strings
+    let result = "";
+    let tokens = text.split(queryRegExp);
+    let tokensLength = tokens.length;
+    for (let i = 0; i < tokensLength; i++) {
+      if (tokens[i].match(tokenRegExp)) {
+        result += '<span background=\"#4a90d9\" foreground=\"#ffffff\">' +
```

Comment:
Do you think an `underline="single"` instead of either of these properties could be enough of a highlight?